lib/std/os/test.zig

```diff
@@ -739,7 +739,7 @@ test "shutdown socket" {
     os.closeSocket(sock);
 }
 
-var signal_test_failed = true;
+var signal_test_handler_called_count: u32 = 0;
```

Comment:
nit: `test_sigaction_count_handler_called` ? This global is only used in `test "sigaction"`, so it might make sense to call it accordingly.

Reply:
It's also the only global `var` in the entire `os/test.zig` file, so might be even better to refactor it out if possible.

I naively tried moving it to inside the `test` block, but I get the `mutable not accessible from here` error that probably prompted the var to be global in the first place.

I could move it into `S` like so:

```diff
     const S = struct {
+        var handler_called_count: u32 = 0;
+
         fn handler(sig: i32, info: *const os.siginfo_t, ctx_ptr: ?*const anyopaque) callconv(.C) void {
             _ = ctx_ptr;
             // Check that we received the correct signal.
             switch (native_os) {
                 .netbsd => {
                     if (sig == os.SIG.USR1 and sig == info.info.signo)
-                        signal_test_handler_called_count += 1;
+                        handler_called_count += 1;
                 },
                 else => {
                     if (sig == os.SIG.USR1 and sig == info.signo)
-                        signal_test_handler_called_count += 1;
+                        handler_called_count += 1;
                 },
             }
         }
```

And the `expect` lines become:

```diff
     try os.raise(os.SIG.USR1);
-    try testing.expect(signal_test_handler_called_count == 1);
+    try testing.expect(S.handler_called_count == 1);
     // Check if the handler has been correctly reset to SIG_DFL
```

Is "namespacing" the var inside the S type like this preferable?  It would remove the global entirely.